kvbc/src/merkle_tree_db_adapter.cpp

```diff
@@ -623,8 +648,7 @@ KeysVector DBAdapter::lastReachableBlockKeyDeletes(BlockId blockId) const {
   // Clear the stale index for the last version.
   add(keysToDelete, staleIndexKeysForVersion(blockNode.stateRootVersion));
 
-  // TODO(DD): Add to keysToDelete the NonProvable keys and NonProvableStale(if any).
-  // key_type|NonProvableStale|stale_since_version|
+  add(keysToDelete, staleIndexKeysForBlock(blockId));
```

Comment:
Can we add a comment about what we are doing and be explicit that the line above adds the tree stale keys (provable) and that one is for non-provable ones?

Reply:
Done